elm-src/.gitignore

```diff
@@ -0,0 +1,3 @@
+elm-stuff/*
+!elm-stuff/exact-dependencies.json
```

Comment:
This is not recommended. Couple of things to keep in mind:
- this file is used to keep track of the entire dependency tree including transitive dependencies
- elm does semver (and does it well)
- if you require version pinning, the recommended approach is by selecting a specific version in `elm-package.json` by restricting the range (i.e. `1.1.1 <= v < 1.1.2` to pin to 1.1.1)

Reply:
So the reason I do this, is so Nix can do a pure build of the application. I'm parsing this file to pre-fetch dependencies at build time. See https://github.com/NixOS/nixos-homepage/pull/160/files#diff-42fbbaa4d2f9ce0b8c95980d6d43cc96R5 to about line 55.

Reply:
Right. Seems weird to bypass elm-package that way, but I don't know how Nix builds work or what the constraints are. I guess what I'm asking is this: what's the added value of this over `elm-package install --yes`?

Reply:
A build under Nix isn't allowed to access the internet unless it can declare ahead of time the contents of what it is going to make.

In other words, there are:

 -  "fixed output" builds (we call them "derivations") where it downloads something from the internet. To have a fixed output derivation, it has to start out by saying what the sha256 of the result is going to be. If what it creates doesn't match the sha256, the build is failed and the result is deleted. In this case, before starting the inputs are unknown (the network!) so the output must be known.
 - standard builds where there is no network access, but the output is not limited. In this case all the inputs are known ahead of time, so the output doesn't have to be known.

This is part of how Nix accomplishes its reproducible build guarantees.

Reply:
Yes, Nix separates download phase from build phase.

Reply:
Right. Got it. I suspect this process could be made a fair bit prettier in 0.19.

Reply:
Indeed - I'm checking every week for an alpha :)